console/frontend/src/main/frontend/src/app/views/status/adapter-status/adapter-status.component.html

```diff
@@ -18,422 +18,479 @@ <h5>
+                      <th>Name</th>
+                      <th>Sender</th>
+                      <th>SlotId</th>
+                      <th></th>
+                    </tr>
+                  </thead>
+                  <tbody>
+                    @for (pipes of adapter.pipes; track pipes.name) {
+                      @if (pipes.sender) {
+                        <tr>
+                          <td>{{ pipes.name }}</td>
+                          <td>
+                            {{ pipes.sender }}
+                            @if (pipes.destination) {
+                              <span #pipeDestination title="{{ pipes.destination }}">
+                                ({{ pipes.destination | truncate: 150 : pipeDestination }})</span
+                              >
+                            }
+                          </td>
+                          <td>{{ pipes.message?.slotId }}</td>
+                          <td>
+                            <span class="pull-right">
+                              <span [ngClass]="pipes.isSenderTransactionalStorage ? 'text-primary' : 'text-success'">
+                                @if (pipes.hasMessageLog) {
+                                  <a
+                                    appHasAccessToLink="browseMessages"
+                                    [routerLink]="[
+                                      '',
+                                      adapter.configuration,
+                                      'adapters',
+                                      adapter.name,
+                                      'pipes',
+                                      pipes.name,
+                                      'stores',
+                                      'Done',
+                                    ]"
+                                    class="btn btn-xs btn-default m-l-xs"
+                                    type="button"
+                                  >
+                                    <i
+                                      [ngClass]="
+                                        pipes.isSenderTransactionalStorage ? 'fa fa-sign-out' : 'fa fa-database'
+                                      "
+                                    ></i>
+                                    ({{ pipes.messageLogCount }})
+                                  </a>
+                                }
+                              </span>
+                            </span>
+                          </td>
+                        </tr>
+                      }
+                    }
+                  </tbody>
+                </table>
+              </div>
+            </div>
+          </div>
+        }
+        <div class="row">
+          <div class="col-lg-9 col-md-12 messages table-responsive">
+            <h4>Messages</h4>
+            <table class="table table-striped" style="min-width: 620px">
+              <tbody>
+                @for (message of adapter.messages; track message.message) {
```

Comment:
Content of a message can be duplicate. I don't really know how it works in this case. But I think it might be better to track the hole message or `$index`, that's a reasonable default anyway.

Reply:
Pretty sure you can't track by $index since that is a variable only exists during for loop iteration and not while during the diffing process, but I could be wrong

Reply:
How interesting the migration set all track properties to the item itself rather than $index, I wonder if that has a reason or if $index is still fine to use

Reply:
Yeah good question, maybe because they don't want to make assumptions, and this way they cover advanced situations as well. I would think using `$index` is more performant when you want to update everything always.

Reply:
Tracking message itself here since its a small and flat object to compare